### PR TITLE
Update Use Cases to include new features

### LINKED ISSUE #191: Update Developer Guide for v1.4

### LINKED ISSUE #214: Manual Testing for sort and find

### PR DESCRIPTION
**Pull request by ShummyOwnzYou**
WIP

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -636,10 +636,32 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+1. User requests to open the help window.
+2. TrackAScholar pops open the help window. <br>
+   Use case ends.
+
+**Extensions**
+
+* 1a. Input format is wrong.
+
+  * 1a1. TrackAScholar displays an error message prompting correct input format. <br>
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md]: what do you mean by correct input format?
If a user keys in Hepl, TrackAsScholar will only display an error message that the command is not recognised?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -656,84 +678,153 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-* Once an applicant is deleted,the applicant will be removed from the database and the user will no longer
-be able to view the applicant from the list.
+
+* An applicant's name cannot be updated to an applicant name that already exists.
 
 **MSS**
 
-1. User request to delete an applicant in TrackAScholar.
-2. TrackAScholar removes applicant with matching name.
+1. User requests to change the details of an applicant (such as email or contact number etc.).
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md]: use ***phone number*** instead of contact number

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -743,38 +834,96 @@ be able to view the applicant from the list.
...
+      Use case resumes at step 1.
 
-* 1b. TrackAScholar detects an error in the entered applicant status/detects no applicant status input.
+* 1b. Input parameters is invalid.
 
-    * 1b1. TrackAScholar requests for the correct applicant status.
-    * 1b2. User enters new applicant status.
-    * Steps 1b1-1b2 are repeated until the applicant status entered matches a valid existing applicant status.
-    * Use case resumes from step 2.
+    * 1b1. TrackAScholar displays an error message showing parameter requirements. <br>
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md:859]: ***parameters***

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -743,38 +834,96 @@ be able to view the applicant from the list.
...
+**Extensions**
+
+* 1a. Input format is wrong.
+
+    * 1a1. TrackAScholar displays an error message prompting correct input format. <br>
+      Use case resumes at step 1.
+
+* 1b. Input parameters is invalid.
+
+    * 1b1. TrackAScholar displays an error message showing parameter requirements. <br>
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md:883]: same as previous

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -743,38 +834,96 @@ be able to view the applicant from the list.
...
+      Use case resumes at step 1.
+
+--------------------------------------------------------------------------------------------------------------------
+
+**Use case: UC11 - Imports a json file into TrackAScholar**
+
+**Guarantees:**
+* After merging, no two applicants will have the same name.
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md]: no two applicants ***in TrackAScholar***

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -743,38 +834,96 @@ be able to view the applicant from the list.
...
+      Use case resumes at step 1.
+
+* 1d. Imported file is not a json file.
+
+    * 1d1. TrackAScholar displays an error message showing the imported file type requirements. <br>
+      Use case resumes at step 1.
+
+* 2a. An imported applicant has the same name as an applicant already in the applicant list.
+
+    * 2a1. TrackAScholar merges the applicants according to the type of merging specified by the user input. <br>
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md:926]: Not sure whether need to specify the type of merging here?

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (ShummyOwnzYou) [reply docs/DeveloperGuide.md:859]: no

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (ShummyOwnzYou) [reply docs/DeveloperGuide.md:883]: also no

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (ShummyOwnzYou) [reply docs/DeveloperGuide.md:926]: not specified as use cases should not go into detail implementation wise

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ShummyOwnzYou) [reply docs/DeveloperGuide.md]: Noted, removed extension. Also removed extension for other relevant use cases.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (ShummyOwnzYou) [reply docs/DeveloperGuide.md]: Removed the brackets as too much detail in implementation.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (ShummyOwnzYou) [reply docs/DeveloperGuide.md]: noted, added improvement.

**Comment by ShummyOwnzYou** [comment]
All use cases completed

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -661,90 +678,194 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-be able to view the applicant from the list.
+
+* An applicant's name cannot be updated to an applicant name that already exists.
 
 **MSS**
 
-1. User request to delete an applicant in TrackAScholar.
-2. TrackAScholar removes applicant with matching name.
+1. User requests to edit the details of an applicant.
+2. TrackAScholar edits the details of the specified applicant to the list. <br>
```
Comment (L1uY1jun) [inline docs/DeveloperGuide.md]: maybe rephrase this as "edits the details of the specified applicant and updates the list"

Comment (ShummyOwnzYou) [reply docs/DeveloperGuide.md]: noted, made improvements

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -1014,6 +1210,29 @@ testers are expected to do more *exploratory* testing.
...
+
+    2. Test case: `sort NAME` <br>
+       Expected: All applicants will be sorted by applicant name in ascending alphabetical order, ignoring case-sensitivity, and will be shown in TrackAScholar GUI.
+   
+    3. Test case: `sort SCHOLARSHIP -r` <br>
+       Expected: All applicants will be sorted by scholarship name in descending alphabetical order, ignoring case-sensitivity, and will be shown in TrackAScholar GUI.
+   
+    4. Test case: `sort PHONE` <br>
+       Expected: No changes displayed on TrackAScholar GUI since TrackAScholar can only sort applicants by `NAME`, `SCHOLARSHIP` and
+       `STATUS` can be filtered. Error details shown in the status message.
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md]: can be filtered?

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -1014,6 +1210,29 @@ testers are expected to do more *exploratory* testing.
...
+
+    5. Other incorrect filter commands to try: `sort` and `sort Alex Yeoh` <br>
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md]: ***sort***

**Comment by L1uY1jun** [comment]
Can add the [Return to top]() icon for all the use cases thxs

File: docs/DeveloperGuide.md (hunk 8 above)
Comment (ShummyOwnzYou) [reply docs/DeveloperGuide.md]: corrected

File: docs/DeveloperGuide.md (hunk 9 above)
Comment (ShummyOwnzYou) [reply docs/DeveloperGuide.md]: corrected

**Comment by benjytan45678** [comment]
LGTM, merged
